auraxium/event/_client.py

```diff
@@ -262,9 +270,12 @@ async def _connection_handler(self) -> None:
         # NOTE: The following "async for" loop will cleanly restart the
         # connection should it go down. Invoking "continue" manually may be
         # used to manually force a reconnect if needed.
-
+        connection_failed = False
```

Comment:
Needing this flag here feels a bit unnecessary given that it should only ever loop on reconnect anyway.

I suppose we could always subscribe first thing after establishing the connection and drop the subscription-queuing from the `add_trigger()` method - would also remove a non-obvious side effect from that method.

I'll make a note to look into that, for the purposes of this PR this is perfectly acceptable.

Reply:
I think you would have to add a check to `add_triger()` to ensure a subscription is still queued if the websocket is already connected, but otherwise that seems like a potentially better way to handle the problem.

Reply:
The other option is just not checking it; subscribing twice doesn't hurt the stream. Anyway, I don't think there is enough of a benefit to flip the responsibilities around there - lgtm!